.github/workflows/ci.yml

```diff
@@ -1,14 +1,20 @@
 name: CI
 
 on:
-  workflow_run:
-    workflows: ["Run tests"] # Waits until the tests worflow is completed, then this workflow starts running
-    types:
-      - completed
+  pull_request:
+    branches: [ main, feat/* ]
+  workflow_dispatch:
 
 jobs:
   coverage:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.11]
+      max-parallel: 1  # This ensures jobs run sequentially, not concurrently
```

Comment:
By running the tests sequentially, we eliminate the occasional nonce collision. Sometimes, we used to get `lower nonce in tx`.